plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java

```diff
@@ -47,8 +50,28 @@ public ImportApplicationWizard() {
 		this.model = new ImportApplicationWizardModel();
 	}
 
+	public ImportApplicationWizard(Map<IProject, Collection<IBuildConfig>> projectsAndBuildConfigs) {
+		this();
+		if (projectsAndBuildConfigs != null && projectsAndBuildConfigs.size() == 1) {
+			Map.Entry<IProject, Collection<IBuildConfig>> entry = projectsAndBuildConfigs.entrySet().iterator().next();
+			IProject project = entry.getKey();
+			Connection connection = ConnectionsRegistryUtil.safeGetConnectionFor(project);
```

Comment:
Presumably this should not be null but my only concern would be the case where the connection is not found; should we use the alternate getter?  Do we need to explicitly handle that case or display something to a user?